File: C/2315-count-asterisks/main.go
package main

func countAsterisks(s string) int {
	cnt := 0
	pipecnt := 0

	for _, b := range s {
		if byte(b) == '*' && pipecnt%2 == 0 {
			cnt++
		}
		if byte(b) == '|' {
			pipecnt++
		}
	}
	return cnt
}

File: C/2315-count-asterisks/main.rs
struct Solution {}

impl Solution {
  pub fn count_asterisks(s: String) -> i32 {
    let mut ret: i32 = 0;
    let mut cnt: i32 = 0;
    for b in s.chars() {
      if b == '|' {
        cnt += 1;
      } else if b == '*' && cnt % 2 == 0 {
        ret += 1;
      }
    }
    ret
  }
}
